File: fonte/u04 Ordenação em memória principal/c/principal.c
#include "bolha.h"
#include "countingsort.h"
#include "heapsort.h"
#include "insercao.h"
#include "mergesort.h"
#include "quicksort.h"
#include "selecao.h"
#include "shellsort.h"

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

//=============================================================================
int main(int argc, char **argv) {

    //Delcaracao de variaveis
    int n = (argc < 2) ? 1000 : atoi(argv[1]);
    int *array = (int*) malloc(n*sizeof(int));
    clock_t inicio, fim;
    double total;


    //Geracao do conjunto a ser ordenado
	 aleatorio(array, n);
	 //crescente(array, n);
	 //decrescente(array, n);


    //Mostrar o conjunto a ser ordenado
	 //mostrar(array, n);
		

    //Execucao do algoritmo de ordenacao
	 inicio = clock();
    //bolha(array, n);
    //countingsort(array, n);
    //heapsort(array, n);
    //insercao(array, n);
    //mergesort(array, n);
    //quicksort(array, n);
    selecao(array, n);
    //shellsort(array, n);
	 fim = clock();
    total = ((fim - inicio) / (double)CLOCKS_PER_SEC);    


    //Mostrar o conjunto ordenado, tempo de execucao e status da ordenacao
	 //algoritmo.mostrar(array, n);
	 printf("Tempo para ordenar: %f s.\n", total);
	 printf("isOrdenado: %s\n", isOrdenado(array, n) ? "true" : "false");

    //Desalocar o espaco de memoria do array
    free(array);

    return 0;
}

File: fonte/u04 Ordenação em memória principal/java/Principal.java
/**
 * Classe Principal
 * @author Max do Val Machado
 * @version 3 08/2020
 */
class Principal {
	public static void main(String[] args) {

      //Delcaracao de variaveis
      Geracao algoritmo;
      int n = (args.length < 1) ? 1000 : Integer.parseInt(args[0]);
      double inicio, fim;


      //Inicializacao do algoritmo de ordenacao
      //algoritmo = new Bolha(n);
      //algoritmo = new Countingsort(n);
      //algoritmo = new Heapsort(n);
      algoritmo = new Insercao(n);
      //algoritmo = new Mergesort(n);
      //algoritmo = new Quicksort(n);
      //algoritmo = new Selecao(n);
      //algoritmo = new Shellsort(n);


      //Geracao do conjunto a ser ordenado
		//algoritmo.aleatorio();
		algoritmo.crescente();
		//algoritmo.decrescente();


      //Mostrar o conjunto a ser ordenado
		//algoritmo.mostrar();
		

      //Execucao do algoritmo de ordenacao
		inicio = algoritmo.now();
		algoritmo.sort();
		fim = algoritmo.now();


      //Mostrar o conjunto ordenado, tempo de execucao e status da ordenacao
		//algoritmo.mostrar();
		System.out.println("Tempo para ordenar: " + (fim-inicio)/1000.0 + " s.");
		System.out.println("isOrdenado: " + algoritmo.isOrdenado());
	}
}
